src/ACEtk/PhotonuclearTable/src/generateData.hpp

```diff
@@ -0,0 +1,136 @@
+    Log::info( "MTR NTR value = {}", mtr.NTR() );
+    Log::info( "LQR NTR value = {}", lqr.NTR() );
+    Log::info( "SIG NTR value = {}", sig.NTR() );
+    throw std::exception();
+  }
+
+  // generate the xss array and set the locators in the jxs array as we go
+  jxs[0] = 1;
+  jxs[1] = eszu.NES() + 1;
+  jxs[2] = eszu.elastic().empty() ? jxs[1] : 0;
+  jxs[3] = eszu.elastic().empty() ? 0 : 2 * eszu.NES() + 1;
+  jxs[4] = eszu.elastic().empty() ? 2 * eszu.NES() + 1 : 3 * eszu.NES() + 1;
+  xss.insert( xss.end(), eszu.begin(), eszu.end() );
+  jxs[5] = xss.size() + 1;
+  xss.insert( xss.end(), mtr.begin(), mtr.end() );
+  jxs[6] = xss.size() + 1;
+  xss.insert( xss.end(), lqr.begin(), lqr.end() );
+  jxs[7] = xss.size() + 1;
+  jxs[8] = xss.size() + ntr + 1;
+  xss.insert( xss.end(), sig.begin(), sig.end() );
+  if ( ptype ) {
+
+
+    std::vector< unsigned int > numbers;
+    for ( const auto& element : mtrh.value() ) {
+
+      numbers.push_back( element.NTR() );
+    }
+
+    auto size = ntype * 12;
+    auto ixs_position = xss.size();
+    std::vector< std::array< unsigned int, 10 > > ixs( ntype );
+
+    jxs[9] = xss.size() + 1;
+    jxs[10] = xss.size() + size + 1;
+
+    for ( std::size_t index = 0; index < ntype; ++index ) {
+
+      auto number = mtrh.value()[index].NTR();
+
+      ixs[index][0] = xss.size() + size + 1;
+      xss.insert( xss.end(), pxs.value()[index].begin(), pxs.value()[index].end() );
+      ixs[index][1] = xss.size() + size + 1;
+      xss.insert( xss.end(), phn.value()[index].begin(), phn.value()[index].end() );
+      ixs[index][2] = xss.size() + size + 1;
+      xss.insert( xss.end(), mtrh.value()[index].begin(), mtrh.value()[index].end() );
+      ixs[index][3] = xss.size() + size + 1;
+      xss.insert( xss.end(), tyrh.value()[index].begin(), tyrh.value()[index].end() );
+      ixs[index][4] = xss.size() + size + 1;
+      ixs[index][5] = xss.size() + size + number + 1;
+      xss.insert( xss.end(), sigh.value()[index].begin(), sigh.value()[index].end() );
+      ixs[index][6] = xss.size() + size + 1;
+      ixs[index][7] = xss.size() + size + number + 1;
+      xss.insert( xss.end(), andh.value()[index].begin(), andh.value()[index].end() );
+      ixs[index][8] = xss.size() + size + 1;
+      ixs[index][9] = xss.size() + size + number + 1;
+      xss.insert( xss.end(), dlwh.value()[index].begin(), dlwh.value()[index].end() );
+    }
+
+    block::IXSU locators( std::move( ptype.value() ), std::move( numbers ),
+                          std::move( ixs ) );
+    xss.insert( xss.begin() + ixs_position, locators.begin(), locators.end() );
+  }
+
+  // set the nxs values for the continuous energy table
+  // NXS(1) = length
+  // NXS(2) = za
+  // NXS(3) = NES (number energies)
+  // NXS(4) = NTR (total number of reactions)
+  // NXS(5) = NTYPE (number of secondary particle types)
+  nxs[0] = xss.size();
+  nxs[1] = z * 1000 + a;
+  nxs[2] = eszu.NES();
+  nxs[3] = ntr;
+  nxs[4] = ntype;
+  nxs[5] = 2;
```

Comment:
Any documentation on what nxs(6, 7, 16) do? As far as I can tell, they're just unreferenced in our code.

Reply:
<img width="878" alt="image" src="https://github.com/njoy/ACEtk/assets/18539456/dc32cbec-dd0c-4882-90f2-bbeb5cad4314">